### PR TITLE
DashR Salesforce CRM

### PR DESCRIPTION
**Pull request by KPhans**
Issue for app: #[issue number here]

# App pull request
- [ ] This is a new app
- [ ] I am improving an existing app (redesigns/code "makeovers")

## About

- Playground deployment URL (new version):
- Current gallery app URL: (delete this line if inapplicable)
- Python app repository link: (delete this line if you are working on a Python app)

## Workflow

- [ ] I have created a branch in the appropriate monorepo, and the
  elements necessary for successful deployment are in place.
- [ ] If the app is a redesigned and/or restyled version of an
  existing gallery app, I've summarized the changes requested in the
  appropriate Streambed issue and confirm that they have been applied.
- [ ] If the app is on the Dash Gallery portal, I have added a link to
  the GitHub repository for the source code in the portal description.
- [ ] If the app is a reimplementation of a Python gallery app for the
  DashR gallery, the app in this PR mimics, as closely as possible,
  the style and functionality of the existing app.=
- [ ] I have removed *all Google Analytics code* from the app's
  `assets/` folder.

## The pre-review review

I have addressed all of the following questions:

- [ ] Does everything in my code serve some purpose? (I have removed
  any dead and/or irrelevant code.)
- [ ] Does everything in my code have a clear purpose? (My code is
  readable and, where it isn't, it has been commented appropriately.)]
- [ ] Am I reinventing the wheel? (I have used appropriate packages to
  lessen the volume of code that needs to be maintained.)

### CODE REVIEW

File: apps/dashr-salesforce-crm/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# dash-salesforce-crm
+This is a demo of the Dash interactive Python framework developed by [Plotly](https://plot.ly/).
+
+Dash abstracts away all of the technologies and protocols required to build an interactive web-based application and is a simple and effective way to bind a user interface around your Python code.
```
Comment (rpkyle) [inline apps/dashr-salesforce-crm/README.md]: Replace Python with R

File: apps/dashr-salesforce-crm/README.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+
+To learn more check out our [documentation](https://plot.ly/dash).
+
+This app uses Salesforce API in order to implement a custom CRM dashboard. The API is used via the module [Simple-Salesforce](https://pypi.org/project/simple-salesforce/). Create a free SalesForce developer trial account: [https://developer.salesforce.com/signup](https://developer.salesforce.com/signup) to utilize this API and run the app.
```
Comment (rpkyle) [inline apps/dashr-salesforce-crm/README.md]: Are we using the same API in R?

File: apps/dashr-salesforce-crm/SFManager.R [hunk 3]
```diff
@@ -0,0 +1,76 @@
+#https://developer.salesforce.com/signup
+#Username = kevinphan@plotly.com, password = Toronto12345%
```
Comment (rpkyle) [inline apps/dashr-salesforce-crm/SFManager.R]: Yikes! Can we swap this out for a publicly available Plotly account?

File: apps/dashr-salesforce-crm/SFManager.R [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+session = session()
+
+getAllFields <- function(objectName) {
+  description <- rforcecom.getObjectDescription(session, objectName)
+  fields <- as.character(description$name)
+  rforcecom.retrieve(session, objectName, fields)
+}
+
+#leads = getAllFields('Lead')
```
Comment (rpkyle) [inline apps/dashr-salesforce-crm/SFManager.R]: :hocho: commented out line?

File: apps/dashr-salesforce-crm/SFManager.R [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  objectName <- "Case"
+  fields <- c('CreatedDate', 'ContactId','Description', 'Type', 'Reason', 'Status', 'Origin', 'Subject', 'Priority', 'IsClosed', 'OwnerId', 'IsDeleted', 'AccountId')
+  return(rforcecom.retrieve(session, objectName, fields))}
+
+get_contacts = function(){
+sf_Manager = setClass('sf_Manager')
+objectName <- "Contact"
+fields <- c('Id', 'Salutation', 'FirstName', 'LastName')
+rforcecom.retrieve(session, objectName, fields)}
+#contacts <- get_contacts()
```
Comment (rpkyle) [inline apps/dashr-salesforce-crm/SFManager.R]: :hocho: commented out line

File: apps/dashr-salesforce-crm/SFManager.R [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+  fields <- c('CreatedDate','CloseDate','Name', 'StageName', 'ExpectedRevenue', 'Amount', 'LeadSource', 'IsWon', 'IsClosed', 'Type', 'Probability')
+  rforcecom.retrieve(session, objectName, fields)}
+
+get_cases = function(){
+  objectName <- "Case"
+  fields <- c('CreatedDate', 'ContactId','Description', 'Type', 'Reason', 'Status', 'Origin', 'Subject', 'Priority', 'IsClosed', 'OwnerId', 'IsDeleted', 'AccountId')
+  return(rforcecom.retrieve(session, objectName, fields))}
+
+get_contacts = function(){
+sf_Manager = setClass('sf_Manager')
```
Comment (rpkyle) [inline apps/dashr-salesforce-crm/SFManager.R]: consider indenting for readability

File: apps/dashr-salesforce-crm/SFManager.R [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+  objectName <- "User"
+  fields <- c('Id', 'FirstName', 'LastName')
+  return(rforcecom.retrieve(session, objectName, fields))}
+
+get_accounts = function(){
+  objectName <- "Account"
+  fields <- c('Id', 'Name')
+  return(rforcecom.retrieve(session, objectName, fields))}
+
+#accounts = get_accounts()
```
Comment (rpkyle) [inline apps/dashr-salesforce-crm/SFManager.R]: :hocho: commented out line

File: apps/dashr-salesforce-crm/init.R [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+remotes::install_github("plotly/dashR", dependencies=FALSE)
+remotes::install_github("plotly/dash-html-components")
+remotes::install_github("plotly/dash-core-components")
+
+
+install.packages('sqldf')
+install.packages('dplyr')
+install.packages('lubridate')
+
```
Comment (rpkyle) [inline apps/dashr-salesforce-crm/init.R]: Do you need a line to install `salesforcer`? And `rjson`?

Comment (KPhans) [reply apps/dashr-salesforce-crm/init.R]: rjson maybe not due to jsonlite but salesforcer yes. IT allows us to access the API

File: apps/dashr-salesforce-crm/README.md (hunk 2 above)
Comment (KPhans) [reply apps/dashr-salesforce-crm/README.md]: yes

**Comment by KPhans** [comment]
@rpkyle made review changes. ready for your review

File: apps/dashr-salesforce-crm/app.R [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+
+app$layout(htmlDiv(
+  list(
+    # header
+    htmlDiv(list(
+      
+      htmlSpan("CRM App using Salesforce API", className='app-title'),
+      
+      htmlDiv(
+        htmlImg(src="https://s3-us-west-1.amazonaws.com/plotly-tutorials/logo/new-branding/dash-logo-by-plotly-stripe.png",height="100%")
```
Comment (rpkyle) [inline apps/dashr-salesforce-crm/app.R]: @KPhans Think you missed this one ☝️

File: apps/dashr-salesforce-crm/functions.R [hunk 10]
```diff
@@ -0,0 +1,76 @@
+library(dashR)
```
[the remark this answers is not in the document]
Comment (rpkyle) [reply apps/dashr-salesforce-crm/functions.R]: @KPhans ☝️ and this one

**Pull request merged**